src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1306,16 +1318,35 @@ private static void ComputeDeclarationsInNode(SemanticModel semanticModel, ISymb
                         executableCodeBlocks = declInNode.ExecutableCodeBlocks;
 
                         // Execute operation actions.
-                        if (shouldExecuteOperationActions && executableCodeBlocks.Any())
+                        if ((shouldExecuteOperationActions || shouldExecuteOperationBlockActions) && executableCodeBlocks.Any())
```

Comment:
I tried it. The method needs 11 parameters for 20 lines of code, and it didn't look like a net win.